### DISCUSSION TITLE
docs/courses/cb&b%20634/2024-09-10/

### DISCUSSION THREAD

**Discussion by giscus** [open-genome/open-genome.github.io#13] [Ideas]
# docs/courses/cb&b%20634/2024-09-10/

A starter template for a Jeykll site using the Just the Docs theme!

https://open-genome.github.io/docs/courses/cb&b%20634/2024-09-10/

**Comment by super-dainiu** [comment]
| Data Structure | Base Size (bytes) | Additional Memory | Notes |
|----------------|------------------|-------------------|--------|
| int | 24 | +4 bytes per additional digit block | Size grows with value magnitude |
| float | 24 | Fixed | IEEE 754 double precision |
| str | Latin-1: 49<br>Other: 74<br>Emoji: 80 | Latin-1: +1 per char<br>Other: +2 per char<br>Emoji: +4 per char | Depends on character encoding |
| list | 56 | +8 per item | Does not include size of contained items |
| array.array | 64 | +8 per value (for floats) | Contiguous memory, single type |
| numpy.array | 96 | +8 per value (for floats) | Contiguous memory, optimized |
| bitarray | 96 | +1 per 8 bits | Efficient for boolean values |
| set | 216 | Nonlinear growth | Empty set base size |
| dict | 232 | Nonlinear growth | Hash table implementation |
| empty class | 16 | Varies by attributes | Base Python object |

Additional notes:
- All Python objects have overhead for reference counting and type information
- Memory usage can vary slightly between Python versions
- Container types (list, set, dict) may allocate extra space for future additions
